BE/app/Http/Services/GuzzleService.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\Response;
+use GuzzleHttp\Exception\GuzzleException;
+
+/**
+ * Class GuzzleService
+ * @package App\Http\Services
+ */
+class GuzzleService
+{
+    private string $base_uri = 'http://universities.hipolabs.com/';
+    private string $endPoint = 'search';
+
+    /**
+     * @return Client
+     */
+    public function newClient(): Client
+    {
+        return new Client(['base_uri' => $this->base_uri]);
+    }
+
+    /**
+     * @return Response|null
+     * @throws GuzzleException
+     */
+    public function sendGetRequest(): ?Response
+    {
+        $client = $this->newClient();
+
+        try {
+            $response = $client->get($this->endPoint, [
+                'query' => [],
+                'verify' => false
+            ]);
+        } catch (\Exception $exception) {
+            $response = null;
```

Comment:
It just was my option, to use response = null instead of exception throwing, I avoided using another one try/catch block in method when I checked this response, and  instead empty response or not. But it is only my option, and here it could be successfully use exception throwing

Reply:
I think it's valid but normally I would also log this.